### PR TITLE
Module 1 - JB Project PR

### PR DESCRIPTION
**Pull request by jaybee1971**
There are still some things I need to do that I think I know how to do, but got stuck on pulling dropdown data values out of the form to update status.  I left comments in the code to what I would do next... will try and do over weekend.

### CODE REVIEW

**Comment by jaybee1971** [comment]
I finally worked it out and now items can be updated and deleted from the list

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Jonesey13) [inline .vscode/settings.json:1]: This file should be in a .gitignore (as `/Users/jb` won't be a folder for most people)

File: app.py [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
+
+@app.route('/create', methods=['POST'])
+def create():
+    session.add_item(request.form['add_todo'])
+    return redirect('/')
+
+
+@app.route('/update', methods=['POST'])
+def update():
+    for id in request.form:
```
Comment (Jonesey13) [inline app.py:24]: This approach limits you to having one input field per todo. You might want to move the `<form>` element into the jinja for loop in `index.html` to make this more extensible in future.

File: templates/index.html [hunk 3]
```diff
@@ -1,18 +1,69 @@
...
   </div>
 
+    <div class="row" style="border:2px solid DodgerBlue;">
+        <div class="col-md-12" style="margin-top: 20px; margin-bottom: 20px;">
+            <h2>Add a New To-Do Item</h2>
+            <form action="/create" method="POST">
+                <div class="form-inline">
+                    <div class="form-group" style="flex: 1">
+                        <label class="col-sm-1" for="add_todo">Title:</label>
+                        <input style="flex: 1" class="form-control" type="text" id="add_todo" name="add_todo" maxlength="64" placeholder="New to-do">
```
Comment (Jonesey13) [inline templates/index.html:18]: Might want to add a `required` attribute here to prevent adding empty todos.

File: templates/index.html [hunk 4]
```diff
@@ -1,18 +1,69 @@
...
-    <div class="col-md-12">
-      <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
+    <div class="col-md-12" style="margin-top: 40px;">
+      <h2 id="todo_table">Your To-Do List</h2>
+        <p>This is a list of your to-do items, check them off once completed or delete as required</p>
+        <form action="/update" method="POST">
+            <div class="form-group">
```
Comment (Jonesey13) [inline templates/index.html:33]: Mix of 2 & 4 space indentation

File: templates/index.html [hunk 5]
```diff
@@ -1,18 +1,69 @@
...
+                    </tr>
+                    </thead>
+                    {% for item in items %}
+                        <tr>
+                            <td>{{item.title}}</td>
+                            <td>{{item.status}}</td>
+                            <td>
+                                <select class="custom-select" name="{{ item.id }}">
+                                    {% if item.status == 'Completed' %}
+                                    <option value="Not Started" name="{{ item.id }}">Not Started</option>
```
Comment (Jonesey13) [inline templates/index.html:49]: These 2 lines should be indented one level in to help show they are part of the `{% if %}` block

File: templates/index.html [hunk 6]
```diff
@@ -1,18 +1,69 @@
...
+                                    <option selected="selected" value="Completed" name="{{ item.id }}">Completed</option>
+                                    {% else %}
+                                    <option selected="selected" value="Not Started" name="{{ item.id }}">Not Started</option>
```
Comment (Jonesey13) [inline templates/index.html:52]: Same here

**Review by Jonesey13 — APPROVED** [review]
Looks all good to me!

Left a few code comments in the review below but feel free to merge this PR in when you are ready 👍

**Pull request merged**